sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/impl/BeamQueryPlanner.java

```diff
@@ -97,6 +102,24 @@
             .typeSystem(connection.getTypeFactory().getTypeSystem())
             .operatorTable(ChainedSqlOperatorTable.of(opTab0, catalogReader))
             .build();
+
+    plannerImplConstructor =
+        getPlannerConstructorFromPipelineOptions(PipelineOptionsFactory.create());
+  }
+
+  public static Constructor getPlannerConstructorFromPipelineOptions(PipelineOptions options) {
+    String plannerImplClassName = null;
+    try {
+      plannerImplClassName = options.as(BeamSqlPipelineOptions.class).getPlannerImplClassName();
+      return Class.forName(plannerImplClassName).getConstructor(FrameworkConfig.class);
```

Comment:
I would probably also support a no-args constructor here. Eventually maybe the one with `pipelineOptions` arg as well.

Reply:
I will not support no-args constructor here as there is no immediate usage. For example, Calcite only has one PlannerImpl with one constructor with FrameworkConfig as a arg [(PlannerImpl).](https://github.com/apache/calcite/blob/master/core/src/main/java/org/apache/calcite/prepare/PlannerImpl.java#L95)

Reply:
We can definitely support more if there is need in the future.